pvanalytics/quality/data_shifts.py

```diff
@@ -0,0 +1,297 @@
+        # Take the median of every day of the year across all years in the
+        # data, and use this as the seasonality of the time series
+        month_values = series.index.month
+        day_values = series.index.day
+        series_seasonality = series_normalized.groupby([month_values,
+                                                        day_values])\
+            .transform("median")
+        # Remove seasonality from the time series
+        return (series_normalized - series_seasonality)
+
+
+def detect_data_shifts(series,
+                       filtering=True, use_default_models=True,
+                       method=None, cost=None, penalty=40):
+    """
+    Detect data shifts in the time series, and return list of dates where these
+    data shifts occur.
+
+    .. warning:: If the passed time series is less than 2 years in length,
+        it will not be corrected for seasonality. Data shift detection will
+        be run on the min-max normalized time series with no seasonality
+        correction.
+
+    Parameters
+    ----------
+    series : Pandas series with datetime index.
+        Daily time series of a PV data stream, which can include irradiance
+        and power data streams. This series represents the summed daily values
+        of the particular data stream.
+    filtering : Boolean, default True.
+        Whether or not to filter out outliers and stale data from the time
+        series. If True, then this data is filtered out before running the
+        data shift detection sequence. If False, this data is not filtered
+        out. Default set to True.
+    use_default_models: Boolean, default True
+        If True, then default change point detection search parameters are
+        used. For time series shorter than 2 years in length, the search
+        function is `rpt.Window`  with `model='rbf'`, `width=50` and
+        `penalty=30`. For time series 2 years or longer in length, the
+        search function is `rpt.BottomUp` with `model='rbf'`
+        and `penalty=40`.
+    method: ruptures search method instance or None, default None.
+        Ruptures search method instance. See
+        https://centre-borelli.github.io/ruptures-docs/user-guide/.
+    cost: str or None, default None
+        Cost function passed to the ruptures changepoint search instance.
+        See https://centre-borelli.github.io/ruptures-docs/user-guide/
+    penalty: int, default 40
+        Penalty value passed to the ruptures changepoint detection method.
+        Default set to 40.
+
+    Returns
+    -------
+    Pandas Series
+        Series of boolean values with a datetime index, where detected
+        changepoints are labeled as True, and all other values are labeled
+        as False.
+
+    References
+    -------
+    .. [1] Perry K., and Muller, M. "Automated shift detection in sensor-based
+       PV power and irradiance time series", 2022 IEEE 48th Photovoltaic
+       Specialists Conference (PVSC). Submitted.
```

Comment:
Can complete this reference now, I think.

Reply:
Updated citation in https://github.com/pvlib/pvanalytics/pull/142/commits/8e0aa881c34223ac4745e7995007169bf1023566